src/offset/local/tzdata.rs

```diff
@@ -0,0 +1,131 @@
+//! This module contains functions to locate and parse the tzdata file, which only
+//! exists on Android and is a database of all tz files.
+//!
+//! Logic was mainly ported from https://android.googlesource.com/platform/libcore/+/jb-mr2-release/luni/src/main/java/libcore/util/ZoneInfoDB.java
+
+use core::{cmp::Ordering, convert::TryInto};
+use std::{
+    fs::File,
+    io::{self, ErrorKind, Read, Seek, SeekFrom},
+};
+
+use super::tz_info::Error;
+
+// The database uses 32-bit (4 byte) integers.
+const TZ_INT_SIZE: usize = 4;
+// The first 12 bytes contain a special version string.
+const MAGIC_SIZE: usize = 12;
+const HEADER_SIZE: usize = MAGIC_SIZE + 3 * TZ_INT_SIZE;
+// The database reserves 40 bytes for each id.
+const TZ_NAME_SIZE: usize = 40;
+const INDEX_ENTRY_SIZE: usize = TZ_NAME_SIZE + 3 * TZ_INT_SIZE;
+
+#[derive(Debug, Clone, Copy)]
+struct Header {
+    index_offset: usize,
+    data_offset: usize,
+    _zonetab_offset: usize,
+}
+
+#[derive(Debug)]
+struct Index(Vec<u8>);
+
+#[derive(Debug, Clone, Copy)]
+struct IndexEntry<'a> {
+    _name: &'a [u8],
+    offset: usize,
+    length: usize,
+    _raw_utc_offset: usize,
+}
+
+pub(super) fn find_tz_data(tz_name: &str) -> Result<Vec<u8>, Error> {
+    let mut file = find_file()?;
```

Comment:
Since this is the only caller for `find_file()`, I think we could just inline it?